<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from all_bot.urdf                   | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="all_bot">
  <material name="black">
    <color rgba="0 0 0 1"/>
  </material>
  <material name="yellow">
    <color rgba="1 1 0 1"/>
  </material>
  <material name="green">
    <color rgba="0 1 0 1"/>
  </material>
  <link name="base_footprint">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0"/>
      <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
  </link>
  <link name="base_link">
    <visual>
      <geometry>
        <box size="0.2413 0.1016 0.0635"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0 "/>
      <material name="black"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.2413 0.1016 0.0635"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0 "/>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.5"/>
      <inertia ixx="0.0005981170833333333" ixy="0.0" ixz="0.0" iyy="0.0025940808333333332" iyz="0.0" izz="0.002856177083333333"/>
    </inertial>
  </link>
  <link name="lidar_link">
    <visual>
      <geometry>
        <cylinder length="0.05715" radius="0.03502"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="green"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.07004 0.07004 0.05715"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.1"/>
      <inertia ixx="5.78776975e-05" ixy="0.0" ixz="0.0" iyy="5.78776975e-05" iyz="0.0" izz="6.132002000000002e-05"/>
    </inertial>
  </link>
  <link name="back_left_wheel">
    <visual>
      <geometry>
        <cylinder length="0.0254" radius="0.03175"/>
      </geometry>
      <origin rpy="1.57 0 0" xyz="0 0 0"/>
      <material name="yellow"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.0254" radius="0.03175"/>
      </geometry>
      <origin rpy="1.57 0 0" xyz="0 0 0"/>
    </collision>
    <inertial>
      <origin rpy="1.57 0 0" xyz="0 0 0"/>
      <mass value="0.1"/>
      <inertia ixx="3.057789583333333e-05" ixy="0.0" ixz="0.0" iyy="3.057789583333333e-05" iyz="0.0" izz="5.040312500000001e-05"/>
    </inertial>
  </link>
  <link name="back_right_wheel">
    <visual>
      <geometry>
        <cylinder length="0.0254" radius="0.03175"/>
      </geometry>
      <origin rpy="1.57 0 0" xyz="0 0 0"/>
      <material name="yellow"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.0254" radius="0.03175"/>
      </geometry>
      <origin rpy="1.57 0 0" xyz="0 0 0"/>
    </collision>
    <inertial>
      <origin rpy="1.57 0 0" xyz="0 0 0"/>
      <mass value="0.1"/>
      <inertia ixx="3.057789583333333e-05" ixy="0.0" ixz="0.0" iyy="3.057789583333333e-05" iyz="0.0" izz="5.040312500000001e-05"/>
    </inertial>
  </link>
  <link name="front_left_wheel">
    <visual>
      <geometry>
        <cylinder length="0.0254" radius="0.03175"/>
      </geometry>
      <origin rpy="1.57 0 0" xyz="0 0 0"/>
      <material name="yellow"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.0254" radius="0.03175"/>
      </geometry>
      <origin rpy="1.57 0 0" xyz="0 0 0"/>
    </collision>
    <inertial>
      <origin rpy="1.57 0 0" xyz="0 0 0"/>
      <mass value="0.1"/>
      <inertia ixx="3.057789583333333e-05" ixy="0.0" ixz="0.0" iyy="3.057789583333333e-05" iyz="0.0" izz="5.040312500000001e-05"/>
    </inertial>
  </link>
  <link name="front_right_wheel">
    <visual>
      <geometry>
        <cylinder length="0.0254" radius="0.03175"/>
      </geometry>
      <origin rpy="1.57 0 0" xyz="0 0 0"/>
      <material name="yellow"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.0254" radius="0.03175"/>
      </geometry>
      <origin rpy="1.57 0 0" xyz="0 0 0"/>
    </collision>
    <inertial>
      <origin rpy="1.57 0 0" xyz="0 0 0"/>
      <mass value="0.1"/>
      <inertia ixx="3.057789583333333e-05" ixy="0.0" ixz="0.0" iyy="3.057789583333333e-05" iyz="0.0" izz="5.040312500000001e-05"/>
    </inertial>
  </link>
  <joint name="base_joint" type="fixed">
    <parent link="base_footprint"/>
    <child link="base_link"/>
    <origin rpy="0 0 0" xyz="0 0 0.07747"/>
  </joint>
  <joint name="base_lidar_joint" type="fixed">
    <parent link="base_link"/>
    <child link="lidar_link"/>
    <origin rpy="0 0 0" xyz="-0.07 0 0.060335"/>
  </joint>
  <joint name="base_back_left_wheel_joint" type="continuous">
    <parent link="base_link"/>
    <child link="back_left_wheel"/>
    <origin rpy="0 0 0" xyz="-0.060335 0.0635 -0.04572"/>
    <axis xyz="0 1 0"/>
  </joint>
  <joint name="base_back_right_wheel_joint" type="continuous">
    <parent link="base_link"/>
    <child link="back_right_wheel"/>
    <origin rpy="0 0 0" xyz="-0.060335 -0.0635 -0.04572"/>
    <axis xyz="0 1 0"/>
  </joint>
  <joint name="base_front_left_wheel_joint" type="continuous">
    <parent link="base_link"/>
    <child link="front_left_wheel"/>
    <origin rpy="0 0 0" xyz="0.060335 0.0635 -0.04572"/>
    <axis xyz="0 1 0"/>
  </joint>
  <joint name="base_front_right_wheel_joint" type="continuous">
    <parent link="base_link"/>
    <child link="front_right_wheel"/>
    <origin rpy="0 0 0" xyz="0.060335 -0.0635 -0.04572"/>
    <axis xyz="0 1 0"/>
  </joint>
</robot>
